frontend/src/metabase/dashboard/hoc/withAutoApplyFilters.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useCallback, useEffect, useState } from "react";
+import _ from "underscore";
+
+import { Dashboard } from "metabase-types/api";
+import { ApplyButton } from "./withAutoApplyFilters.styled";
+
+type ParameterValues = Record<string, any>;
+
+interface WithAutoApplyFiltersProps {
+  dashboard: Dashboard;
+  parameterValues: ParameterValues;
+  setParameterValue: (id: string, value: any) => void;
+  setParameterValues: (parameterValues: ParameterValues) => void;
+  fetchDashboardCardData: ({
+    reload,
+    clear,
+    ignoreCache,
+  }: {
+    reload?: boolean;
+    clear?: boolean;
+    ignoreCache?: boolean;
+  }) => void;
+}
+
+export default function withAutoApplyFilters<T>(
```

Comment:
It seems that this approach goes against the current one where `redux` is used to store dashboard-wide data. Instead of introducing another approach, could you move it to `redux` instead, alongside other dashboard reducers, actions, and selectors?

Reply:
That's great advice. I haven't thought of that initially. I'll do that 👍